### ISSUE TITLE
it took a long time to start training

### ISSUE THREAD

**Issue by NoNameCanUse9** [ultralytics/ultralytics#9271]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

hello guys ,I found the time to prepare training suddenly turns long  and  stuck it  on this step  ,but  i  have not change any seeting on my  pc.  i runs  it in wsl.
rain: Scanning /home/choken/deeplearning/train.cache... 14996 images, 97 backgrounds, 3 corrupt: 100%|██████████| 14996/14996 [00:00<?, ?it/s]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_1.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0125]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_2.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0075]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_8.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0225]
val: Scanning /home/choken/deeplearning/train.cache... 14996 images, 97 backgrounds, 3 corrupt: 100%|██████████| 14996/14996 [00:00<?, ?it/s]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_1.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0125]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_2.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0075]
train: WARNING ⚠️ /home/choken/deeplearning/train/002926_8.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0225]
Plotting labels to runs/detect/default_nopretrain4/labels.jpg... 

### Environment

Ultralytics YOLOv8.1.2 🚀 Python-3.10.13 torch-2.1.0+cu121 CUDA:0 (NVIDIA GeForce RTX 4060 Ti, 16380MiB)
Setup complete ✅ (12 CPUs, 31.9 GB RAM, 126.3/476.9 GB disk)

### Additional

from ultralytics import YOLO as yolo
import torch
    
    # 你的设备和模型加载代码
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

model = yolo('yolov8n.yaml')  # or yolov5m, yolov5l, yolov5x, custom

model.train(data='./data.yaml', epochs=300, batch=55,  device=0,resume=False,name="default_nopretrain",pretrained=False	)  # train 300 epochs

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by NoNameCanUse9** [comment]
sometimes it stucked so long that the training was crash

**Comment by Burhan-Q** [comment]
@NoNameCanUse9 your labels are not properly formatted, which is why you're getting errors like this:

`non-normalized or out of bounds coordinates [ 1.0225]`

See the [dataset docs](https://docs.ultralytics.com/datasets/detect/) for guidance on how to properly format your annotations. The `x y w h` values (all four) need to be within the bounds $[0.0, 1.0]$ and $1.0 < 1.0225$ which will cause issues. If the docs don't make sense, you should try downloading the `coco8` example dataset and reference how the annotations are formatted.

**Comment by NoNameCanUse9** [comment]
> @NoNameCanUse9 your labels are not properly formatted, which is why you're getting errors like this:
> 
> `non-normalized or out of bounds coordinates [ 1.0225]`
> 
> See the [dataset docs](https://docs.ultralytics.com/datasets/detect/) for guidance on how to properly format your annotations. The `x y w h` values (all four) need to be within the bounds [0.0,1.0] and 1.0<1.0225 which will cause issues. If the docs don't make sense, you should try downloading the `coco8` example dataset and reference how the annotations are formatted.

I get it。But i  have been using this dataset for a long time .this step only took about ten seconds before. In few days ago。I found it  ·Stuck here  about ten minutes。So I'm very confused

**Comment by glenn-jocher** [comment]
Hey @NoNameCanUse9 👋! It's quite peculiar that this issue has suddenly appeared, especially if you've been using the same dataset without any problems until now. One possibility might be changes or updates in your environment or the Ultralytics YOLO version affecting how data is processed. 🤔 Could you try clearing any cached data files and rerunning your training command? Sometimes, a fresh start can help resolve such anomalies. Additionally, ensure your Ultralytics YOLO version hasn't recently updated, or if it has, review the update notes for any changes related to dataset processing. 

Let's see if this helps:

```bash
rm -rf /home/choken/deeplearning/train.cache  # Clear cache
```

Then rerun your training command. Keep us posted on how it goes!

**Comment by glenn-jocher** [comment]
@NoNameCanUse9 hey there! 🎉 Glad to hear clearing the cache worked and you can proceed. Regarding the warnings about corrupt images/labels with non-normalized/out-of-bounds coordinates, it seems these specific images still have annotation issues. Even though you've been using this dataset without problem, slight changes in the Ultralytics YOLO version might cause previously unnoticed issues to surface. 🕵️‍♂️

Here's a quick fix to check and correct these labels in Python:
```python
def check_labels(file_path):
    with open(file_path, 'r') as f:
        labels = [x.split() for x in f.read().strip().split('\n')]
    for label in labels:
        coords = [float(x) for x in label[1:]]
        if not all(0.0 <= x <= 1.0 for x in coords):
            print(f"Label out of bounds found: {file_path}")
            # Optionally, correct the label here
            # coords = [max(0.0, min(1.0, x)) for x in coords]

# Use it like this for each label file
check_labels('/path/to/your/label/file.txt')
```

It's great to hear the optimizer adjustments are being automatically handled! 👍 This ensures your training is using the best possible settings for your setup. Keep an eye on those TensorBoard visuals for insights into how your model is learning. If you have more issues or questions, feel free to shout out. Happy training! 🚀
